lib/environment.js

```diff
@@ -32,6 +35,9 @@ exports.Environment = Environment = function(options) {
    delete options.tags;
    delete options.debug;
    this.config = options;
+   
+   this.resourcesDir = this.loader && typeof(this.loader.findResourcesDir) === "function" ? this.loader.findResourcesDir() : "./resources";
```

Comment:
I looked at how django and rails do translation resource lookup, and I think we don't have to provide any new funtionality (findResourceDir et al.) to keep the advantages you mention.

This is how I imagine it:

In a multi-app scenario, each app would configure its template environment with its own resource path (a relative path within the app's package). The translation resources would be distributed - per application - but not shared. A super-app could not override sub-app translations but on the other hand the sub-apps wouldn't even have to agree on one translation package (e.g.: ilib and gettext).

This would still leave the disadvantage of dealing with distributed translations. Django has rosetta [1] to make that easier. I think this problem is out-of-scope for our first implementation.

[1] http://django-rosetta.readthedocs.org/en/latest/usage.html#generate-a-batch-of-files-to-translate

somewhat related: A location tool extracting the translatable strings would also operate per application: given a list of file extensions and paths, the extractor runs the files through a new subclass of parser, which only cares about the translation nodes and will output the skeleton translation resources.